src/main/java/io/github/debuggyteam/tablesaw/TableSawScreenHandler.java

```diff
@@ -94,35 +98,78 @@ public ItemStack quickTransfer(PlayerEntity player, int fromIndex) {
 		if (slot != null && slot.hasStack()) {
 			ItemStack initial = slot.getStack();
 			result = initial.copy();
-			if (fromIndex == 0) {
-				if (!this.insertItem(result, 2, 38, false)) {
+			
+			// Descriptive constants used here where possible. Could still use a total rewrite but the behavior is good.
+			
+			// Note: insertItem's "to" param is [seemingly] intentionally off-by-one. Thanks Mojang.
+			// To insert into only slot 2, DO:
+			//     insertItem(result, 2, 3, false)
+			// DO NOT:
+			//     insertItem(result, 2, 2, false)
+			// For this reason you'll see a lot of + 1's for clarity.
+			
+			if (fromIndex == INPUT_SLOT) {
+				// Try to move the item from the input slot back to the inventory, or failing that, the hotbar.
+				
+				if (!this.insertItem(result, FIRST_INVENTORY_SLOT, LAST_HOTBAR_SLOT + 1, false)) {
 					return ItemStack.EMPTY;
 				}
-			} else if (fromIndex == 1) {
-				if (!this.insertItem(result, 2, 38, false)) {
+			} else if (fromIndex == OUTPUT_SLOT) {
+				// Try to move the item from the output slot into the inventory, or the hotbar if no inventory slots are available.
+				
+				if (!this.insertItem(result, FIRST_INVENTORY_SLOT, LAST_HOTBAR_SLOT + 1, false)) {
 					return ItemStack.EMPTY;
 				}
-			} else if (fromIndex >= 2 && fromIndex < 29) {
-				if (!this.insertItem(result, 29, 38, false)) {
-					return ItemStack.EMPTY;
+			} else if (fromIndex >=FIRST_INVENTORY_SLOT && fromIndex <= LAST_HOTBAR_SLOT) {
+				// Item is coming from the inventory or hotbar, so try putting it into the input slot first.
+				
+				if (this.insertItem(result, INPUT_SLOT, INPUT_SLOT + 1, false)) {
+					// Success!
+					if (result.isEmpty()) {
+						//Very important to prevent dupes
+						slot.setStack(ItemStack.EMPTY);
+					}
+					
+					System.out.println("Count remaining: "+result.getCount()+" initial: "+initial.getCount());
```

Comment:
`System.out.println("Count remaining: "+result.getCount()+" initial: "+initial.getCount());` probably looks better as `System.out.println("Count remaining: " + result.getCount() + " initial: " + initial.getCount());`

Reply:
Sounds like I left a debug message in there by accident

Reply:
everyone's done that at least once lol